app/telemetry/telemetry.android.js

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {NativeModules} from 'react-native';
+import LocalConfig from 'assets/config'; // eslint-disable-line
+
+import {store} from 'app/mattermost';
+
+import {
+    saveToTelemetryServer,
+    getDeviceInfo,
+    setTraceRecord,
+} from './telemetry_utils';
+
+const {StartTime} = NativeModules;
+
+class Telemetry {
+    constructor() {
+        this.appStartTime = 0;
+        this.reactInitializedStartTime = 0;
+        this.reactInitializedEndTime = 0;
+        this.metrics = [];
+        this.currentMetrics = {};
+        this.pendingSinceLaunchMetrics = [];
+
+        this.initialize();
+    }
+
+    async initialize() {
+        if (StartTime) {
+            const nativeTimes = await StartTime.getNativeTimes();
+            this.appStartTime = nativeTimes.appStartTime;
+        }
+    }
+
+    reset() {
+        this.metrics = [];
+        this.currentMetrics = {};
+        this.pendingSinceLaunchMetrics = [];
+    }
+
+    canSendTelemetry() {
+        const {config} = store.getState().entities.general;
+        return Boolean(!__DEV__ && config.EnableDiagnostics === 'true' && LocalConfig.TelemetryEnabled);
```

Comment:
Do we want to include that the native module is available as well?

Reply:
For me, it's no need. Since we're updating telemetry config via build, I think check to `LocalConfig.TelemetryEnabled` is sufficient enough.